src/biogeochem/dynHarvestMod.F90

```diff
@@ -173,6 +180,61 @@ subroutine dynHarvest_interp(bounds)
   end subroutine dynHarvest_interp
 
 
+  !-----------------------------------------------------------------------
+  subroutine dynHarvest_interp_resolve_harvesttypes(bounds, harvest_rates, after_start_of_harvest_ts)
+    !
+    ! !DESCRIPTION:
+    ! Get harvest data for model time, when needed.
+    !
+    ! Note that harvest data are stored as rates (not weights) and so time interpolation
+    ! is not necessary - the harvest rate is held constant through the year.  This is
+    ! consistent with the treatment of changing PFT weights, where interpolation of the
+    ! annual endpoint weights leads to a constant rate of change in PFT weight through the
+    ! year, with abrupt changes in the rate at annual boundaries.
+    !
+    ! Note the difference between this and dynHarvest_interp is that here, we keep the different
+    ! forcing sets distinct (e.g., for passing to FATES which has distinct primary and secondary lands)
+    ! and thus store it in harvest_typeresolved
+    !
+    ! !USES:
+    use dynTimeInfoMod , only : time_info_type
+    !
+    ! !ARGUMENTS:
+    type(bounds_type), intent(in) :: bounds  ! proc-level bounds
+    real(r8)         , intent(out):: harvest_rates(bounds%begg: , 1: ) ! output the harvest rates
+    logical          , intent(out):: after_start_of_harvest_ts
+    !
+    ! !LOCAL VARIABLES:
+    integer               :: varnum       ! counter for harvest variables
+    real(r8), allocatable :: this_data(:) ! data for a single harvest variable
+
+    character(len=*), parameter :: subname = 'dynHarvest_interp'
+    !-----------------------------------------------------------------------
+
+    !!! NOTE I AM NOT USING THE BOUNDS_LEVEL_PROC ASSERTION HERE BECAUSE OF WHERE IN THE CALLING
```

Comment:
bounds is going to be whatever type of bounds that was sent into it. The caller needs to determine what type of bounds is sent down. I would remove the language that specifies what type of bounds this is. It looks like the caller is actually a clump level bounds, but this routine doesn't need to know what type it is.

Reply:
this one too?

Reply:
This function does require specific proc-level bounds, otherwise it will not work correctly because the harvest bounds are at a different level than the clumping, site, column, etc. The caller should currently be passing these bounds correctly, the comment in the type declaration line notes the required bounds from the caller. I am not sure about the line starting with "!!!..." as I think it was there from before.